autoendpoint/src/error.rs

```diff
@@ -152,6 +152,58 @@ impl ApiErrorKind {
         }
     }
 
+    /// Specify the label to use for metrics reporting.
+    pub fn metric_label<'a>(&self) -> &'a str {
+        match self {
+            ApiErrorKind::PayloadError(_) => "payload_error",
+            ApiErrorKind::Router(_) => "router",
+
+            ApiErrorKind::Validation(_) => "validation",
+            ApiErrorKind::InvalidEncryption(_) => "invalid_encryption",
+            ApiErrorKind::NoTTL => "no_ttl",
+            ApiErrorKind::InvalidRouterType => "invalid_router_type",
+            ApiErrorKind::InvalidRouterToken => "invalid_router_token",
+            ApiErrorKind::InvalidMessageId => "invalid_message_id",
+
+            ApiErrorKind::VapidError(_) => "vapid_error",
+            ApiErrorKind::Jwt(_) => "jwt",
+            ApiErrorKind::TokenHashValidation(_) => "token_hash_validation",
+            ApiErrorKind::InvalidAuthentication => "invalid_authentication",
+            ApiErrorKind::InvalidLocalAuth(_) => "invalid_local_auth",
+
+            ApiErrorKind::InvalidToken => "invalid_token",
+            ApiErrorKind::InvalidApiVersion => "invalid_api_version",
+
+            ApiErrorKind::NoUser => "no_user",
+            ApiErrorKind::NoSubscription => "no_subscription",
+
+            ApiErrorKind::LogCheck => "log_check",
+
+            ApiErrorKind::Io(_) => "io",
+            ApiErrorKind::Metrics(_) => "metrics",
+            ApiErrorKind::Database(_) => "database",
+            ApiErrorKind::EndpointUrl(_) => "endpoint_url",
+            ApiErrorKind::RegistrationSecretHash(_) => "registration_secret_hash",
+        }
+    }
+
+    /// Don't report all errors to sentry
+    pub fn is_reportable(&self) -> bool {
```

Comment:
We've discussed this before -- our old `reportable` naming is a bit confusing here and we should use something else. I called this `is_sentry_event` in contile.

Reply:
Ah, right. Forgot about that. I'll change it.